### PR TITLE
Add Image Hub sample app for Consul

### PR DESCRIPTION
**Pull request by kanishkarj**
We might want to test it after getting the custom copy of consul on the cluster. But yes the resources do get deployed.

Addresses issue #58 

@leecalcote, @kushthedude, @nveenjain do check! 

Signed-off-by: kanishkarj <kanishkarj@hotmail.com>

### CODE REVIEW

**Comment by kanishkarj** [comment]
I have verified it by deploying it on the custom consul that we are using, It works perfectly. Though I am not sure how to expose the ingress service, via NodePort or LoadBalancer? @leecalcote

**Comment by leecalcote** [comment]
> I have verified it by deploying it on the custom consul that we are using, It works perfectly. Though I am not sure how to expose the ingress service, via NodePort or LoadBalancer? @leecalcote

Let's go ahead and optimize for local deployments and experimentation in local deployments; let's use `NodePort` to facilitate.

File: consul/config_templates/image-hub.yaml [hunk 1]
```diff
@@ -100,9 +100,9 @@ metadata:
   labels:
     app: envoy
 spec:
-  # type: LoadBalancer
+  type: LoadBalancer
```
[the remark this answers is not in the document]
Comment (kanishkarj) [reply consul/config_templates/image-hub.yaml]: I have removed it now.

[the remark this answers is not in the document]
Comment (kanishkarj) [reply consul/config_templates/image-hub.yaml]: I have changed NodePort as you have pointed out above.

**Comment by kanishkarj** [comment]
Sure! Done.

**Pull request merged**